### PR TITLE
fix #566 change the append

### LINKED ISSUE #566: bug about build lock key
- [ ] I have searched the [issues](https://github.com/alibaba/fescar/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
the flag in the pic doesn't work! and user  Stringbuilder instead of Stringbuffer
image

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version :
- OS :
- Others:

### PR DESCRIPTION
**Pull request by github-ygy**
### Ⅰ. Describe what this PR did
change the append And replace Stringbuffer to StringBuild when thread safe

### CODE REVIEW

File: rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/ConnectionContext.java [hunk 1]
```diff
@@ -16,13 +16,13 @@
 
 package com.alibaba.fescar.rm.datasource;
 
+import com.alibaba.fescar.common.exception.ShouldNeverHappenException;
+import com.alibaba.fescar.rm.datasource.undo.SQLUndoLog;
+
```
Comment (jovany-wang) [inline rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/ConnectionContext.java]: remove line21.

File: rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/exec/BaseTransactionalExecutor.java [hunk 2]
```diff
@@ -31,6 +27,10 @@
 import com.alibaba.fescar.rm.datasource.sql.struct.TableRecords;
 import com.alibaba.fescar.rm.datasource.undo.SQLUndoLog;
```
Comment (jovany-wang) [inline rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/exec/BaseTransactionalExecutor.java]: Also remove this line

File: rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/exec/BaseTransactionalExecutor.java [hunk 3]
```diff
@@ -201,15 +201,12 @@ protected String buildLockKey(TableRecords rowsIncludingPK) {
         StringBuilder sb = new StringBuilder();
         sb.append(rowsIncludingPK.getTableMeta().getTableName());
         sb.append(":");
-
-        boolean flag = false;
+        int filedSequence = 0;
         for (Field field : rowsIncludingPK.pkRows()) {
-            if (flag) {
+            sb.append(field.getValue());
+            if (++filedSequence < rowsIncludingPK.pkRows().size()) {
```
Comment (jovany-wang) [inline rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/exec/BaseTransactionalExecutor.java]:
```java
sb.append(field.getValue());
filedSequence++;
 if (filedSequence < rowsIncludingPK.pkRows().size()) {
```
Better to not lead the if statement too complex.

File: rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/ConnectionContext.java (hunk 1 above)
Comment (github-ygy) [reply rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/ConnectionContext.java]: done

File: rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/exec/BaseTransactionalExecutor.java (hunk 2 above)
Comment (github-ygy) [reply rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/exec/BaseTransactionalExecutor.java]: done

File: rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/exec/BaseTransactionalExecutor.java (hunk 3 above)
Comment (github-ygy) [reply rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/exec/BaseTransactionalExecutor.java]: accepted

**Comment by github-ygy** [comment]
thanks @slievrly @jovany-wang ，all comments is accepted。

File: rm-datasource/src/test/java/com/alibaba/fescar/rm/datasource/exec/BaseTransactionalExecutorTest.java [hunk 4]
```diff
@@ -16,16 +16,24 @@
...
+import com.alibaba.fescar.rm.datasource.sql.struct.TableRecords;
 import org.junit.Assert;
 import org.junit.Test;
 
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import static org.mockito.Mockito.*;
```
Comment (slievrly) [inline rm-datasource/src/test/java/com/alibaba/fescar/rm/datasource/exec/BaseTransactionalExecutorTest.java]: why import *?

**Comment by slievrly** [comment]
@jovany-wang LGTM

**Pull request merged**